### ISSUE TITLE
lsp-mode + rust-analyzer do not pick up changes made to Cargo.toml on MacOS

### ISSUE THREAD

**Issue by jbr** [emacs-lsp/lsp-mode#3370]
I'm not sure if this is a bug report, feature request, or a configuration problem on my end. I've been writing rust with rust-analyzer and lsp-mode for a while now and have gotten accustomed to the following sequence every time I add a crate:

1. either edit Cargo.toml in emacs or use cargo-edit (`cargo add` / `cargo rm`) from the command line
2. `killall rust-analyzer`
3. type `y` to the lsp-mode prompt asking if I want to restart the server, for each open rust-analyzer workspace (sometimes quite a few)

Ideally, rust-analyzer + lsp-mode would detect a change to Cargo.toml and either restart the server or pick up the new configuration. If that's not possible for some reason, it would be really nice and probably fairly straightforward to add a `lsp-restart-server` function which restarts just the server for the current lsp workspace, skipping the confirmation prompt. If there's an lsp command to tell the server to reload the workspace without killing the process, an lsp-mode binding to that would be another convenient solution.

Thanks, and please let me know if this works as expected for other people and I can try to figure out what's misconfigured on my end.

Environment, if relevant:
MacOS 12.2.1, lsp-mode 20220223.639, latest rust-analyzer (built from git daily)
EmacsGNU Emacs 27.2 (build 1, x86_64-apple-darwin20.6.0, Carbon Version 164 AppKit 2022.6) of 2021-11-16

**Comment by jadestrong** [comment]
Related issue https://github.com/emacs-lsp/lsp-mode/issues/3296

You can manually execute the `lsp-rust-analyzer-reload-workspace` method to reload the workspace.
After some troubleshooting, this is because the backend `kqueue` used by `filenotify.el` on Mac does not support the editing of files under the watching folder. `fsevents` is a good choice on Mac, but emacs still doesn't support it.

**Comment by jadestrong** [comment]
If your `create-lockfiles` variable is `nil`, you can try set `(setq create-lockfiles t)` . In my case, which can make it works as expected .

**Comment by jbr** [comment]
Does `create-lockfiles` fix the problem with `cargo-edit` (`cargo add`)?

**Comment by jadestrong** [comment]
No,  `cargo-edit` or edit `Cargo.toml` in vscode which doesn't create a `.#' lock file,  so these operations will not trigger listening.
